### PR TITLE
USAGOV-25 - Federal Directory Design

### PR DESCRIPTION
**Pull request by thetypicalprogrammer**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-25

## Description
This is the initial implementation of the Federal Directory.  The scope of this branch got away from us, although I'm  (akf) not sure how we could have split it up. 

There's a doc in this branch at docs/Federal_Directory.md that includes an overview and setup steps (the latter are duplicated in the Jira ticket). 

## Checklist for the Developer

- [x] A link to the JIRA ticket has been added above.
- [x] The JIRA ticket identifies the desired result of this change.
- [ ] The JIRA ticket contains clear acceptance criteria.
- [x] The JIRA ticket contains clear testing steps.
- [ ] The JIRA ticket contains a description of "Done".
- [x] Any preparation/installation/update steps required for this code to work properly (drush commands, scripts, configuration updates) are documented in the ticket.

## Checklist for the Peer Reviewers
- [x] The branch name of this PR matches the project standards.
- [ ] QA steps are followed and any changes to process are updated in the ticket.
- [ ] Code Standards are followed, there are no bad practices in use.
- [ ] Config changes (if any) include only necessary changes.
- [ ] No errors in Drupal or Client Browser.
- [ ] Code has been tested locally (if possible).
- [ ] Following AC and Testing Steps verify changes work as expected.
- [ ] There are no known side-effects outside of expected behavior.

### CODE REVIEW

**Comment by akf** [comment]
Some of the more recent work on this request SHOULD NOT be merged into dev.

**Comment by oadekoya12** [comment]
No Worries. Let me know when it is ready

File: config/sync/s3fs.settings.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ _core:
 credentials_file: ''
 bucket: ''
 region: us-gov-west-1
-use_cname: true
+use_cname: false
```
Comment (akf) [inline config/sync/s3fs.settings.yml]: We probably don't mean to change this file.

Comment (akf) [reply config/sync/s3fs.settings.yml]: Just FYI, I'm going through this and looking for files appear to be changing by accident, or that we committed but are no longer using. I'm using the "review" to mark things so I can easily find them and correct them in a bit.

File: config/sync/block.block.leftmenuenglish.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+uuid: 8637c0ae-eb33-4d64-9006-7482c54dbc74
```
Comment (akf) [inline config/sync/block.block.leftmenuenglish.yml]: This file can be deleted.

File: config/sync/block.block.pagetitle.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+uuid: 780f7045-714c-4dea-b367-e312f9092b01
```
Comment (akf) [inline config/sync/block.block.pagetitle.yml]: Delete

**Review by ednark — CHANGES_REQUESTED** [review]
web/sites/default/settings.php accidentally reverts a bunch of changes. I don't see any of your changes in here so you can safely just override this file from current dev.

File: config/sync/field.field.node.directory_record.field_government_branch.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+uuid: c5a1418f-c82a-4996-8972-985906ef82f4
```
Comment (akf) [inline config/sync/field.field.node.directory_record.field_government_branch.yml:1]: Make sure this is still showing a pick list of options

**Review by ednark — CHANGES_REQUESTED** [review]
web/themes/custom/usagov/templates/menu--sidebar_first.html.twig. We can't leave in the node id check. We will have to convert it to something more generic. Worst case we need to add in a whole new field to track some flag that only this one node will use. Node ID should be stable for at least a little while, so if we HAVE to do this for time constraint reasons we can do this as long as we have an immediate technical debt ticket to follow up with a permanent solution.

File: config/sync/field.storage.node.field_offices_near_you.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+uuid: be6405b5-8dc1-41ef-8ff4-84b0353dc8b9
```
Comment (akf) [inline config/sync/field.storage.node.field_offices_near_you.yml]: Still need to replace this with a link (multiple values accepted)

File: config/sync/field.storage.node.field_synonym.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+uuid: c7e1ddd8-3a74-4485-bbd7-18b0db8f8314
```
Comment (akf) [inline config/sync/field.storage.node.field_synonym.yml:1]: Remove (until we figure out how to make these work with the index)

File: config/sync/system.site.yml [hunk 7]
```diff
@@ -11,4 +11,4 @@ page:
   front: /node/1
 admin_compact_mode: false
 weight_select_max: 100
-default_langcode: en
+default_langcode: en
```
Comment (akf) [inline config/sync/system.site.yml]: We also don't mean to change this. It's just a newline, but it clutters the PR.

**Comment by akf** [comment]
I'll remove that menu--sidebar check; we don't need it if we don't need the sidebar next to the individual directory record pages. We just need to *not* tell the directory records to have menu entries.

File: config/sync/field.field.node.directory_record.field_tty_.yml [hunk 8]
```diff
@@ -0,0 +1,21 @@
+uuid: ad1246c1-4f24-4b5c-a8a0-44ff46d5ba5e
```
Comment (akf) [inline config/sync/field.field.node.directory_record.field_tty_.yml]: field_tty_ is a weird name; maybe fix before it gets out in the wild

File: config/sync/views.view.locked_content.yml [hunk 9]
```diff
@@ -17,120 +17,12 @@ base_table: node_field_data
 base_field: nid
```
Comment (akf) [inline config/sync/views.view.locked_content.yml]: Revert; we don't mean to change this.

**Review by ednark — CHANGES_REQUESTED** [review]
web/themes/custom/usagov/templates/views-view-summary.html.twig
there appears to be an errant close svg tag. it looks like someone tried to comment it out but missed the last line.

File: web/sites/default/settings.php [hunk 10]
```diff
@@ -797,26 +797,6 @@
  * Cloud Foundry places all service credentials in VCAP_SERVICES
```
Comment (akf) [inline web/sites/default/settings.php]: Revert; not an intentional change

**Review by akf — CHANGES_REQUESTED** [review]
This is all stuff for me to do.

**Review by ednark — CHANGES_REQUESTED** [review]
web/themes/custom/usagov/templates/node--directory-record--full.html.twig
Should not use tables for layout, only tabular data. This should be turned into divs or something.

**Review by ednark — CHANGES_REQUESTED** [review]
web/themes/custom/usagov/html-blocks/final_fed-directory.html
web/themes/custom/usagov/html-blocks/directory_detail.html
I don't think these should be declaring a doctype? since they are just html chunks

**Comment by akf** [comment]
> web/themes/custom/usagov/templates/menu--sidebar_first.html.twig. We can't leave in the node id check. We will have to convert it to something more generic. Worst case we need to add in a whole new field to track some flag that only this one node will use. Node ID should be stable for at least a little while, so if we HAVE to do this for time constraint reasons we can do this as long as we have an immediate technical debt ticket to follow up with a permanent solution.

Okay, I'm taking this code out (and I'm in full agreement on that). I have re-discovered two other issues I have when I turn off "show in menu" for the directory record nodes: 
* The Nav menu renders with out any entries (so we see a blue horizontal bar and the layout is weird). This is easily fixed by suppressing that block for pages under /federal-agencies. 
* No breadcrumbs. :-( 

There'll be a followup ticket on this; I'm just noting it here as well.

**Comment by akf** [comment]
Brainwave: I can suppress the menu expansion when the number of nodes "below" the current entry exceeds a certain number, like 50. I definitely need to run this by UX, but it actually makes logical sense.

**Comment by akf** [comment]
> web/themes/custom/usagov/templates/node--directory-record--full.html.twig
> Should not use tables for layout, only tabular data. This should be turned into divs or something.

@kellyyan @thetypicalprogrammer Please take another look at this. In addition to the table issue, there's probably an accessibility guideline to take into account here. If you look at https://www.usa.gov/federal-agencies/u-s-department-of-agriculture you can see divs being used there for the layout and H3 markup for the labels. (There's a lot of other stuff going on there that, as far as I know, we don't want to carry over.)  Hopefully USWDS provides some guidance on doing the zebra shading for a series of divs.

**Comment by thetypicalprogrammer** [comment]
> > web/themes/custom/usagov/templates/node--directory-record--full.html.twig
> > Should not use tables for layout, only tabular data. This should be turned into divs or something.
> 
> @kellyyan @thetypicalprogrammer Please take another look at this. In addition to the table issue, there's probably an accessibility guideline to take into account here. If you look at https://www.usa.gov/federal-agencies/u-s-department-of-agriculture you can see divs being used there for the layout and H3 markup for the labels. (There's a lot of other stuff going on there that, as far as I know, we don't want to carry over.) Hopefully USWDS provides some guidance on doing the zebra shading for a series of divs.

This should now be resolved in the latest commits

**Review by akf — DISMISSED** [review]

**Comment by akf** [comment]
@ednark I think we've addressed all your comments. Thank you for reviewing!

**Review by ednark — APPROVED** [review]

**Pull request merged**